clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -340,15 +340,14 @@ bool ProgramInfo::link() {
       // If there was a checked type on a variable in the input program, it
       // should stay that way. Otherwise, we shouldn't be adding a checked type
       // to an extern function.
-      if (!G->getReturnVar()->getIsOriginallyChecked()) {
-        std::string Rsn = "Return value of an external function:" + FuncName;
+      std::string Rsn =
+        "Unchecked pointer in parameter or return of external function " +
+        FuncName;
+      if (!G->getReturnVar()->getIsGeneric())
```

Comment:
Why this exception? I guess if it's generic, we are assuming the underlying pointer, even if `void *`, should not be constrained to WILD?